pixeltable/exprs/type_cast.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import Optional, Dict, List, Tuple, Any
+
+import sqlalchemy as sql
+
+from pixeltable.exprs import DataRow, Expr, RowBuilder
+from pixeltable.type_system import ColumnType
+
+
+class TypeCast(Expr):
+
+    def __init__(self, underlying: Expr, new_type: ColumnType):
+        super().__init__(new_type)
+        if self.col_type.is_string_type() \
+                or self.col_type.is_json_type() and underlying.col_type.is_string_type():
+            # It's a valid type conversion
+            self.components = [underlying]
+            self.id = self._create_id
```

Comment:
This needs to be 'id = self._create_id()'

I'm not sure how to catch this other than to run a type checker (which we should look into; my guess is we'll get a ton of false positives).

Also, this is missing coverage by the serialization test (test_exprs:test_serialization).

Reply:
Oops ...

I'm surprised this worked at all, it's such an egregious type violation :/

I added some type hints to the initializer